src/components/disappearing-header/index.tsx

```diff
@@ -183,11 +190,35 @@ const DisappearingHeader: React.FC<Props> = ({
 
   const onScrolling = useCallback(
     (e: NativeScrollEvent) => {
+      const scrollPos = e.contentOffset.y;
+      scrollYValue.current = scrollPos;
       endReachListener(e);
     },
-    [endReachListener],
+    [endReachListener, scrollYValue],
   );
 
+  useEffect(() => {
+    if (!tabPressBehaviour) return;
+    const {navigation, onTabPressOnTopScroll} = tabPressBehaviour;
+
+    const unsubscribe = navigation
+      .dangerouslyGetParent()
+      // Typescript doesn't know that tabLongPress exist in the parent
+      // and types aren't properly exposed to compensate for that.
+      ?.addListener('tabPress' as any, () => {
+        if (!navigation.isFocused()) return;
+        if (!isScrolledToTop(scrollYValue.current, contentHeight)) return;
+        onTabPressOnTopScroll();
+      });
+
+    return unsubscribe;
+  }, [
+    tabPressBehaviour?.navigation,
+    tabPressBehaviour?.onTabPressOnTopScroll,
+    scrollYValue,
```

Comment:
Bruker ref her for å unngå ny subscriber for hver scroll men fortsatt ha gjeldende korrekt scroll data.